app.rb

```diff
@@ -11,24 +12,24 @@ class MakersBnB < Sinatra::Base
   end
 
   get '/' do
-    @email = session[:email]
-    @password = session[:password]
     erb :index
   end
 
-  get '/users/new' do
-    erb :'users/new'
-  end
+  get '/sessions/new' do
+    @valid = session[:valid]
+    erb :'sessions/new'
+  end 
 
-  post '/users' do
-    redirect '/'
+  post '/users/new' do
+    User.create({ email: params['email address'], password: params['password'] })
+    redirect '/spaces'
   end
-  
-  post '/sessions' do
-    session[:email] = params['email']
-    session[:password] = params['password']
-    'authenticate'
-    redirect '/'
+
+  post '/sessions/new' do
+    user = User.where({ email: params['email address'], password: params['password'] }).first
+    redirect '/sessions/new' unless user
+    session[:valid] = true
```

Comment:
if else statements should be in the controller. Maybe better to use a flash message.